test/e2e/storage/csi_mock/csi_honor_pv_reclaim_policy.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi_mock
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	storagehelpers "k8s.io/component-helpers/storage/volume"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epv "k8s.io/kubernetes/test/e2e/framework/pv"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"k8s.io/utils/ptr"
+)
+
+var _ = utils.SIGDescribe("CSI Mock honor pv reclaim policy", func() {
```

Comment:
If we changes as below, we need to find a job that can test this feature. now, there're two PRs that tries to resolve the similar problem. 

- https://github.com/kubernetes/test-infra/pull/31844
- https://github.com/kubernetes/test-infra/pull/31717

```suggestion
var _ = utils.SIGDescribe("CSI Mock honor pv reclaim policy", feature. HonorPVReclaimPolicy, func() {
```

Reply:
/cc @msau42